### PR TITLE
Fix client race condition in logger test

### PR DESCRIPTION
**Pull request by alexandraorth**

### CODE REVIEW

**Comment by alexandraorth** [comment]
@bolerio @aelred Please review this, you've both experienced this bug.

**Comment by alexandraorth** [comment]
I'm going to look for 5 travis passes on this one.

**Comment by alexandraorth** [comment]
travis pass #1

**Comment by alexandraorth** [comment]
please retest this please - `ai.grakn.test.engine.controller.CommitLogControllerTest.whenCommittingGraph_TaskManagerReceivesCountTask`

**Comment by alexandraorth** [comment]
travis pass #2

**Comment by alexandraorth** [comment]
travis pass #3

File: grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java [hunk 1]
```diff
@@ -71,17 +73,28 @@ public void setupSession(){
...
-    public void whenSingleQueryLoadedAndTaskCompletionFunctionThrowsError_ErrorIsLogged(){
+    public void whenSingleQueryLoadedAndTaskCompletionFunctionThrowsError_ErrorIsLogged() throws InterruptedException {
+        CountDownLatch callbackCompleted = new CountDownLatch(1);
+
         // Create a BatchMutatorClient with a callback that will fail
         BatchMutatorClient loader = loader();
-        loader.setTaskCompletionConsumer((json) -> assertTrue("Testing Log failure",false));
+        loader.setTaskCompletionConsumer((json) -> {
+            try {
+                fail("Testing failures appear in logs");
```
Comment (bolerio) [inline grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java]: Why is that a failure? What happens when a Assert.fail is called asynchronously? Does it actually fail the test?

File: grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java [hunk 2]
```diff
@@ -71,17 +73,28 @@ public void setupSession(){
...
 
         // Load some queries
         generate(this::query).limit(1).forEach(loader::add);
 
         // Wait for queries to finish
         loader.waitToFinish();
 
+        // Wait for callback function to execute - Callbacks are run asynchronously
+        callbackCompleted.await(10, TimeUnit.SECONDS);
+
```
Comment (bolerio) [inline grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java]: I think you should check if callbackCompleted.await returns 'false' and fail the test if hat happens. You are expecting it to complete in 10s, but if it doesn't that would reveal a bug somewhere.

File: grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java (hunk 1 above)
Comment (alexandraorth) [inline grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java]: It does not fail the test. Its the equivalent of throwing an `AssertionError`

File: grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java (hunk 2 above)
Comment (alexandraorth) [inline grakn-test/src/test/java/ai/grakn/client/BatchMutatorClientTest.java]: Yep, good idea, will do.

**Comment by alexandraorth** [comment]
travis pass #4

**Comment by alexandraorth** [comment]
@aelred Im not sure we want to do that. I don't think I've well explained this test explained in the comments. I only want the callback function to fail so that we can test the failure is logged.

**Comment by alexandraorth** [comment]
travis pass #5

**Comment by aelred** [comment]
@alexandraorth yes, that's clearer! Maybe include a message like "deliberate failure" or something.

**Comment by alexandraorth** [comment]
@aelred done.

**Pull request merged**